src/js/background.js

```diff
@@ -691,14 +670,26 @@ Badger.prototype = {
   },
 
   /**
-   * Add the tracker and action to the tab.trackers object in tabData
-   * which will be used by the privacy badger popup
+   * Add the third party origin and action to the tabData[tabId] object
+   * so it can be used for the popup and the badge count.
+   *
    * @param tabId the tab we are on
-   * @param fqdn the tracker to add
+   * @param fqdn the third party origin to add
    * @param action the action we are taking
+   *
+   * @returns {boolean} true if badge count should be incremented
    **/
-  logTrackerOnTab: function(tabId, fqdn, action) {
-    this.tabData[tabId].trackers[fqdn] = action;
+  logThirdPartyOriginOnTab: function(tabId, fqdn, action) {
+    this.tabData[tabId].origins[fqdn] = action;
+
+    if (constants.BLOCKED_ACTIONS.hasOwnProperty(action)) {
+      if (!this.tabData[tabId].blocked[action].hasOwnProperty(fqdn)) {
+        this.tabData[tabId].blocked[action][fqdn] = true;
+        this.tabData[tabId].blockedCount += 1;
+        return true;
```

Comment:
if `blocked` is removed this could be:
```javascript
if (!this.tabData[tabId].origins.hasOwnProperty(fqdn)) {
  this.tabData[tabId].origins[fqdn] = action;
  if (!constants.BLOCKED_ACTIONS.has(action)) {
    this.tabData[tabId].blockedCount += 1;
  }
}
```